acceptance/tests/base/dsl/helpers/host_helpers_test.rb

```diff
@@ -0,0 +1,1377 @@
+require "fileutils"
+
+# NOTE: Currently scp failures throw a Net::SCP::Error exception in SSH connection
+# close, which ends up not being caught properly, and which ultimately results
+# in a RuntimeError. The SSH Connection is left in an unusable state and all
+# later remote commands will hang indefinitely.
+#
+# TODO: fix via: https://tickets.puppetlabs.com/browse/BKR-464
+def test_scp_error_on_close?
+  !!ENV["TEST_SCP_ERROR_ON_CLOSE"]
+end
+
+# NOTE: currently there is an issue with the tmpdir_on helper on cygwin
+#       platforms:  the `chown` command always fails with an error about not
+#       recognizing the Administrator:Administrator user/group.  Until this is
+#       fixed, we add this shim that delegates to a non-`chown`-executing version
+#       for the purposes of our test setup.
+#
+# TODO: fix via: https://tickets.puppetlabs.com/browse/BKR-496
+def tmpdir_on(hosts, path_prefix = '', user=nil)
+  return create_tmpdir_on(hosts, path_prefix, user) unless Array(hosts).first.is_cygwin?
```

Comment:
I've moved to `confine_block` (which actually doesn't work, see #944 for a fix) in all my conditionalized tests.  This helper (`tmpdir_on`) is basically a shim to make it possible to write tests using remote temp directories (which are necessarily for test isolation), in light of bug BKR-496.  If we close BKR-496 I can remove this shim.